src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,20 +1,70 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
-  @GetMapping("/greeting")
-  public String greeting(
-      @RequestParam(name = "name", required = false, defaultValue = "World") String name,
-      Model model) {
-    model.addAttribute("name", name);
-    return "greeting";
+  @Autowired
+  private BookRepository bookRepository;
+
+  @GetMapping
+  public Iterable findAll() {
+    return bookRepository.findAll();
+  }
+
+  @GetMapping("/title/{bookTitle}")
+  public List findByTitle(@PathVariable String bookTitle) {
```

Comment:
We can return an `List<Book>` instead of `List`, to be more specific about our returning type

Reply:
Besides that, to be more Restful we can use a `@RequestParam` instead of `@PathParam`. That way, giving that our resources are the books and not the titles, we can use this endpoint as `GET /api/books?title=ATitle` instead of `GET /api/books/title/ATitle`